drivers/bridge/setup_ip_tables.go

```diff
@@ -48,16 +58,25 @@ func setupIPChains(config *configuration) (*iptables.ChainInfo, *iptables.ChainI
 		}
 	}()
 
-	isolationChain, err := iptables.NewChain(IsolationChain, iptables.Filter, false)
+	isolationChain1, err := iptables.NewChain(IsolationChain1, iptables.Filter, false)
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("failed to create FILTER isolation chain: %v", err)
+		return nil, nil, nil, nil, fmt.Errorf("failed to create FILTER isolation chain: %v", err)
 	}
 
```

Comment:
If creating isolationChain2 fails, this code will not roll back the creation of isolationChain1.  Please add a deferred cleanup function as above w/ the DockerChains for Filter and NAT.  Also, although the original code didn't have a cleanup for isolationChain (now isolationChain1), please add one to gracefully handle failure to add the return rule2.

Reply:
done